File: flight_price_analysis/v1/get/iOS SDK/FlightPriceAnalysis.swift
import Amadeus

var amadeus: Amadeus = Amadeus(
    client_id: "YOUR_API_ID",
    client_secret: "YOUR_API_SECRET"
)

amadeus.analytics.itineraryPriceMetrics.get(params: ["originIataCode": "MAD",
    "destinationIataCode": "CDG",
    "departureDate": "2022-01-15"],
    onCompletion: { result in
        switch result {
        case let .success(response):
            print(response.data)
        case let .failure(error):
            fatalError(error.localizedDescription)
        }
    })

File: flight_price_analysis/v1/get/Node SDK/flight_price_analysis.js
const Amadeus = require("amadeus");

const amadeus = new Amadeus({
  clientId: "YOUR_API_KEY",
  clientSecret: "YOUR_API_SECRET",
});

async function main() {
  try {
    // Am I getting a good deal on this flight?
    const response = await amadeus.analytics.itineraryPriceMetrics.get({
      originIataCode: "MAD",
      destinationIataCode: "CDG",
      departureDate: "2022-01-13",
    });

    console.log(response);
  } catch (error) {
    console.error(error);
  }
}

main();
